### ISSUE TITLE
Rethink block only tcp and udp protocol and not blocking other protocols

### ISSUE THREAD

**Issue by rajagopalan181** [celzero/rethink-app#1803]
Rethink block only tcp and udp protocol and not blocking other protocols as per this list ( https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml )

**Comment by ignoramous** [comment]
> and not blocking other protocols as per this list ( https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml )

These other protocols are seldom used on clients like smartphones. And regardless, not handling these protocols does not leak them but blackholes them (at least from what I understand from how VPN tunnels are setup on Android).

Also see: The scope of this project ([as put up in the readme](https://github.com/celzero/rethink-app/blob/cd9a6392b51d1cc3b0ad3dd7f5dcd043df7b29c0/README.md)):

> Rethink doesn't aim to be a feature-rich traditional firewall: It is more in-line with [Little Snitch](https://www.obdev.at/products/littlesnitch/index.html) than IP tables, say.

And on [our subreddit r/rethinkdns](https://www.reddit.com/r/rethinkdns/comments/1envlxo/proxy_vs_vpn/):

> tldr; Rethink does *not* leak packets or break encryption by running WireGuard as a "Proxy".
> 
> ---
>
> A VPN is analogus to a [Router](https://en.wikipedia.org/wiki/Router_(computing)). Operates at layer3 ([L3](https://en.wikipedia.org/wiki/Network_layer)) on Internet Protocol (IP) packets.
>
> A Proxy is analogus to a [Load Balancer](https://en.wikipedia.org/wiki/Load_balancing_(computing)#Load_balancer_features). Operates at [L4](https://en.wikipedia.org/wiki/Transport_layer) (layer4) on Transport Protocol (ex: TCP / UDP) packets.
>
> A Firewall can be of two kinds: [Stateful](https://en.wikipedia.org/wiki/Stateful_firewall) and [Stateless](https://en.wikipedia.org/wiki/Firewall_(computing)#Packet_filter) (packet filter).
>
> In Rethink's context, the app creates a VPN (L3); then proceeds to handle [TCP](https://en.wikipedia.org/wiki/Transmission_Control_Protocol) (L4), [UDP](https://en.wikipedia.org/wiki/User_Datagram_Protocol) (L4), and [ICMP](https://en.wikipedia.org/wiki/Internet_Control_Message_Protocol) (L3 over L4) protocols *statefully*. Rethink does not "understand" / keep track of the state of other protocols; and *drops* (blocks) them (which isn't the same as "leaking" them). Most apps (99.9%, if I were to hazard a guess) work fine with either UDP *or* TCP and rarely need or use other Transport protocols. P2P / LAN apps might do some funny things, and so *may* not work out-of-the-box when Rethink is running.
>
> WireGuard works at L3, but not in Rethink. After applying IP / domain firewall rules, the packets are [statefully transformed](https://en.wikipedia.org/wiki/Network_socket#Sockets_in_network_equipment) (either to an L4 connection-oriented *stream* for TCP, or to a L4 connectionless *datagram* for *both* UDP & ICMP) & forwarded to (any of the mutliple, simulataneously running) WireGuard or underlying network. The protocols Rethink does not deal with are dropped. Today, due to limitations in Android (lack of "ownership" details on ICMP datagrams), ICMP is *always* routed to the underlying network (usually, mobile and/or wifi).

> Rethink's setup may seem restrictive or peculiar but it works nicely for Android.
